roofit/roofitcore/src/RooVectorDataStore.cxx

```diff
@@ -1301,16 +1302,18 @@ void RooVectorDataStore::recalculateCache( const RooArgSet *projectedArgs, Int_t
 {
   if (!_cache) return ;
 
-  pRealVector tv[1000] ;
+  if (static_cast<Int_t>(_cacheUpdateList.size()) < _cache->_nReal) {
```

Comment:
The way the code worked in the beginning was to directly access the elements of the array. I fixed the array overflow by introducing as few changes as possible: As it works now, reserve won't do the job, it would need more changes further below.